### PR TITLE
deploy from a fork using a workflow

### PR DESCRIPTION
**Pull request by jakecoffman**
Since fork PRs can only get read access from GITHUB_TOKEN, we can't use the `docker-branch-releases.yml` workflow to deploy a fork to GHCR before merging it.

This works around that with a manually dispatched workflow that pulls in the PR code, builds, and deploys it. Since it was kicked off by the team it _should_ have the correct permissions.

> **Note**
> While implementing this, I tried to simply `git merge main` after the `gh pr checkout` but git errors with `fatal: refusing  to merge unrelated histories`. So I assume the `gh` command is pulling in the PR as a different tree. To work around that, I squashed the changes and cherry picked them over to main. If anyone has a better or easier solution to that let me know!

### CODE REVIEW

File: .github/workflows/docker-fork-releases.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Check if pull request is approved
+        # The PR must be approved. This is mostly to ensure you've typed the correct PR.
+        # Note: forks will have a blank review decision without approval. Not NEEDS_REVIEW.
+        run: |
+          DECISION=$(gh pr view ${{ github.event.inputs.pr }} --json reviewDecision -t {{.reviewDecision}})
+          echo "Review decision is: $DECISION"
+          [[ $DECISION == "APPROVED" ]] || exit 1
```
Comment (Nishnha) [inline .github/workflows/docker-fork-releases.yml:34]: Does this exit the job or just the step?

File: .github/workflows/docker-fork-releases.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    permissions:
+      contents: read
+      packages: write
+    env:
+      TAG: ${{ github.sha }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Check if pull request is approved
```
Comment (Nishnha) [inline .github/workflows/docker-fork-releases.yml:28]: Is this step necessary?
see https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#about-workflow-runs-from-public-forks

> To help prevent this, workflows on pull requests to public repositories from some outside contributors will not run automatically, and might need to be approved first. By default, all first-time contributors require approval to run workflows.

Comment (Nishnha) [reply .github/workflows/docker-fork-releases.yml:28]: Ah the workflow approval is different from an approval on a PR. Disregard

File: .github/workflows/docker-fork-releases.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            .
+
+      - name: Build dependabot-updater image
+        env:
+          DOCKER_BUILDKIT: 1
+        run: |
+          docker build \
+            -t "$UPDATER_IMAGE:$TAG" \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from "$BASE_IMAGE" \
```
Comment (Nishnha) [inline .github/workflows/docker-fork-releases.yml:67]: cache from ubuntu:20.04 seems a bit strange. Any reason we include this line?

File: .github/workflows/docker-fork-releases.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            --cache-from "$UPDATER_IMAGE_MIRROR" \
```
Comment (Nishnha) [inline .github/workflows/docker-fork-releases.yml:68]: double checked that this is public at https://github.com/dependabot/dependabot-core/pkgs/container/dependabot-updater

File: .github/workflows/docker-fork-releases.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            --build-arg OMNIBUS_VERSION=$TAG \
```
Comment (Nishnha) [inline .github/workflows/docker-fork-releases.yml:69]: This really just pulls the tagged GHCR `dependabot/dependabot-core` image in https://github.com/dependabot/dependabot-core/blob/main/Dockerfile.updater#L2

**Review by Nishnha — APPROVED** [review]

File: .github/workflows/docker-fork-releases.yml (hunk 1 above)
Comment (jakecoffman) [reply .github/workflows/docker-fork-releases.yml:34]: This will fail the workflow.

File: .github/workflows/docker-fork-releases.yml (hunk 3 above)
Comment (jakecoffman) [reply .github/workflows/docker-fork-releases.yml:67]: Yeah I agree, I copied this from another workflow.

File: .github/workflows/docker-fork-releases.yml (hunk 5 above)
Comment (jakecoffman) [reply .github/workflows/docker-fork-releases.yml:69]: Well, uses the image built in the `Build dependabot-updater image` step above, but yes!

**Pull request merged**